puzzle-utils/words/wordsearch.py

```diff
@@ -23,25 +23,32 @@ def wordsearch_reduce(grid, words):
     """
     results = wordsearch_find(grid, words)
 
+    # Create a normalized grid for reference and another for keeping track of used cells
+    reference_grid = []
     working_grid = []
     for row in grid:
-        working_grid.append(list(row))
+        reference_row = [x.lower() for x in list(row)]
+        reference_grid.append(reference_row)
+        working_grid.append(reference_row[:])
 
+    # Remove words from the working grid
     for result in results:
         row = result.row
         column = result.column
         row_increment, column_increment = direction_to_increment_mapping[result.direction]
-        for _ in result.word:
+        while len(result.word) > 0:
```

Comment:
Out of curiosity, why the change in the type of iteration? This way you have to count the length and do a string replace each loop